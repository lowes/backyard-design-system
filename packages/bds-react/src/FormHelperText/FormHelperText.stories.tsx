import React from 'react'
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs'
import Info from '@lowes-tech/bds-icons/InfoOutlined'
import { ApiLink } from '../utils/storybook/ApiLink'
import {
    GridV3 as Grid,
    FormControl,
    TextField,
    FormHelperText,
    TextFieldSkeleton,
    FormHelperTextSkeleton,
} from '../'

export default { title: '@lowes-tech/bds-react/FormHelperText', decorators: [withKnobs] }

const states = [
    'State',
    {
        undefined: undefined,
        error: 'error',
    },
    undefined,
] as const

export const DefaultFormHelperText = () => (
    <Grid.Column start={6} end={12}>
        <FormControl state={select(...states)}>
            <TextField id='test-1' />
            <FormHelperText>{text('Helper Text', 'Placeholder')}</FormHelperText>
        </FormControl>

        <ApiLink to='#' />
    </Grid.Column>
)

export const WithIcon = () => (
    <Grid.Column start={6} end={12}>
        <FormControl>
            <TextField id='test-1' />
            <FormHelperText icon={<Info />}>{text('Helper Text', 'Placeholder')}</FormHelperText>
        </FormControl>

        <ApiLink to='#' />
    </Grid.Column>
)

export const Skeleton = () => (
    <Grid.Column start={6} end={12}>
        <FormControl>
            <TextFieldSkeleton animate={boolean('Animate', false)} />
            <FormHelperTextSkeleton animate={boolean('Animate', false)} />
        </FormControl>
    </Grid.Column>
)
