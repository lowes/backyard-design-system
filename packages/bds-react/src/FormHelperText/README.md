# @lowes-tech/bds-react/FormHelperText

## Interfaces

- [FormHelperTextProps](interfaces/FormHelperTextProps.md)

## References

### default

Renames and re-exports [FormHelperText](README.md#formhelpertext)

## Type aliases

### FormHelperTextRef

Ƭ **FormHelperTextRef**: `HTMLSpanElement`

## Variables

### FormHelperText

• **FormHelperText**: `BDSForwardRef`<[`FormHelperTextProps`](interfaces/FormHelperTextProps.md)\>

Backyard React Helper Text

 <FormControl>
     <TextField />
     <FormHelperText>Example Helper Text</FormHelperText>
 </FormControl>

Text to display as helper with other components.
Used with <FormControl> to correctly indent from input `indentHelper`
