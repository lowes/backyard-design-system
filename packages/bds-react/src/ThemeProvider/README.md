# @lowes-tech/bds-react/ThemeProvider

## Namespaces

- [ThemeProvider](modules/ThemeProvider.md)

## Interfaces

- [BackyardTheme](interfaces/BackyardTheme.md)
- [ThemeProviderProps](interfaces/ThemeProviderProps.md)

## References

### default

Renames and re-exports [ThemeProvider](README.md#themeprovider)

## Type aliases

### BaseTheme

Ƭ **BaseTheme**: typeof `Theme`

## Variables

### ThemeConsumer

• **ThemeConsumer**: `React.Consumer`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeConsumer`

ThemeConsumer same as 'styled-components'

___

### ThemeContext

• **ThemeContext**: `React.Context`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeContext`

ThemeContext same as 'styled-components'

___

### ThemeProvider

• **ThemeProvider**: `BDSFunctional`<[`ThemeProviderProps`](interfaces/ThemeProviderProps.md)\>

Backyard React Theme Provider

 <ThemeProvider theme="light" font="fellix" someOtherContext="value">
     <Grid.Container>
         ...
             <Button />
         ...
     </Grid.Container>
 </ThemeProvider>

Custom Theme Provider wrapper to provide extra context and overrides for Backyard Themes.

## Functions

### dropShadow

▸ `Const` **dropShadow**(`shadow`, `defaultShadow?`): `string`

Converts box-shadow to drop-shadow, if necessary.

Use for `filter` css instead of `box-shadow`.

**`example`**
```
dropShadow(theme.shadows.shadow_04)
 => 'drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.2)) 
         drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.14)) 
         drop-shadow(0px 1px 3px rgba(0, 0, 0, 0.12))'
```

#### Parameters

| Name | Type | Default value | Description |
| :------ | :------ | :------ | :------ |
| `shadow` | `string` | `undefined` | box-shadow |
| `defaultShadow` | `string` | `'none'` | default shadow if none given |

#### Returns

`string`

___

### getShape

▸ `Const` **getShape**(`shapeProp`, `shapeContext`): ``"rounded"`` \| ``"squared"``

Get the shape or shape context in priority

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shapeProp` | `ExtendedShapes` | shape value from props |
| `shapeContext` | ``"rounded"`` \| ``"squared"`` | shape value from context |

#### Returns

``"rounded"`` \| ``"squared"``

___

### splitShadows

▸ `Const` **splitShadows**(`shadows`, `defaultShadow`): `string`[]

Split the shadows in a given string

**`example`**
```
splitShadows('0px 2px 1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12)')
 => [
     '0px 2px 1px rgba(0, 0, 0, 0.2)',
     '0px 1px 1px rgba(0, 0, 0, 0.14)',
     '0px 1px 3px rgba(0, 0, 0, 0.12)'
 ]
```

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shadows` | `string` | shadow string from backyard |
| `defaultShadow` | `string` | default shadow if none given |

#### Returns

`string`[]

___

### useBackyardTheme

▸ `Const` **useBackyardTheme**(`props?`): [`BackyardTheme`](interfaces/BackyardTheme.md)

#### Parameters

| Name | Type |
| :------ | :------ |
| `props` | `UseBackyardThemeProps` |

#### Returns

[`BackyardTheme`](interfaces/BackyardTheme.md)

___

### validateBackyardTheme

▸ `Const` **validateBackyardTheme**(`theme`, `name`): `boolean`

#### Parameters

| Name | Type |
| :------ | :------ |
| `theme` | [`BackyardTheme`](interfaces/BackyardTheme.md) |
| `name` | `string` |

#### Returns

`boolean`
