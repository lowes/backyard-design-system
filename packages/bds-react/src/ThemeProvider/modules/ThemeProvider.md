# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
