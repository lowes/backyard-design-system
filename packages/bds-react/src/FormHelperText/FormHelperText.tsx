import * as React from 'react'
import classNames from 'classnames'

import type { PathIconProps } from '@lowes-tech/bds-icons/components/PathIcon'

import type { BackyardBaseProps, BackyardToken } from '../utils/typings/BackyardProps'

import Typography from '../Typography'
import type { TypographyProps } from '../Typography'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'
import { useBackyardTheme, validateBackyardTheme } from '../ThemeProvider'
import { useFormControl } from '../FormControl'

import FormHelperTextWrapper from './styles/FormHelperTextWrapper'

/**
 * Backyard React Helper Text
 *
 *  <FormControl>
 *      <TextField />
 *      <FormHelperText>Example Helper Text</FormHelperText>
 *  </FormControl>
 *
 * Text to display as helper with other components.
 * Used with <FormControl> to correctly indent from input `indentHelper`
 */
const FormHelperText: BDSForwardRef<FormHelperTextProps> = React.forwardRef<
    FormHelperTextRef,
    FormHelperTextProps
>(function FormHelperText(
    {
        children = 'Helper',
        indent: indentProp,
        icon,
        disabled: disabledProp,
        state: stateProp,
        className,
        typographyProps,
        style,
        ...props
    },
    ref,
) {
    // Get Backyard Theme Context
    const theme = useBackyardTheme({ validate: true, name: 'FormHelperText' })

    validateBackyardTheme(theme, 'FormHelperText')

    let { disabled, state, indent } = useFormControl({ noContext: true })

    // Props override calculated values
    if (indentProp !== undefined) indent = indentProp
    if (disabledProp !== undefined) disabled = disabledProp
    if (stateProp !== undefined) state = stateProp

    // Default values
    if (indent === undefined) indent = 'size_16'

    return (
        <FormHelperTextWrapper
            className={classNames('helper-text', state, { disabled, indent }, className)}
            style={style}
            {...props}
            ref={ref}
        >
            <Typography
                variant="helper"
                {...typographyProps}
                style={{
                    ...(indent ? {
                        paddingLeft:
                            typeof indent === 'string' ? theme.sizes[indent] : theme.sizes.size_16,
                    } : {}),
                    ...typographyProps?.style,
                }}
            >
                {React.isValidElement(icon)
                    ? React.cloneElement(icon, { className: 'helper-text-icon', size: 'size_12' })
                    : null}
                {children}
            </Typography>
        </FormHelperTextWrapper>
    )
})

type FormHelperTextRef = HTMLSpanElement

interface FormHelperTextProps extends BackyardBaseProps<FormHelperTextRef> {
    /**
     * String to use with <Typography>
     */
    children: string
    /**
     * Override indent appearance
     */
    indent?: keyof BackyardToken['sizes'] | boolean
    /**
     * Icon to appear before text
     */
    icon?: React.ReactElement<PathIconProps>
    /**
     * Override disabled appearance
     */
    disabled?: boolean
    /**
     * Override state appearance
     */
    state?: 'default' | 'success' | 'error' | 'info' | 'warning'
    /**
     * DOM Class Name
     */
    className?: string
    /**
     * Typography Props to extend Typography child with
     */
    typographyProps?: React.PropsWithRef<TypographyProps>
}

FormHelperText.bdsName = 'FormHelperText'

export { FormHelperText }

export type { FormHelperTextProps, FormHelperTextRef }

export default FormHelperText
