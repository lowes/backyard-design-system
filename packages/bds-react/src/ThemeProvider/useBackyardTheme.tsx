import * as React from 'react'

import { ThemeContext } from './ThemeProvider'
import type { BackyardTheme } from './ThemeProvider'

const validateBackyardTheme = (theme: BackyardTheme, name: BackyardTheme['name']) => {
    if (!theme?.name) {
        // eslint-disable-next-line no-console
        throw Error(
            `Backyard: A Backyard component ${
                name ? `(\`${name}\`)` : ''
            } is missing a Backyard Theme.\n` +
                'This component uses Backyard tokens and a Backyard Theme is required.\n' +
                `Please wrap ${
                    name ? `\`${name}\` or` : ''
                } your entire application in a Backyard \`ThemeProvider\`.\n` +
                'For a getting started guide, see \n' +
                '`https://backyard.lowes.com/GetStarted/Develop#create-theme-provider`',
        )
    }

    return true
}

const useBackyardTheme = (props: UseBackyardThemeProps = {}) => {
    const { validate = false, name }: UseBackyardThemeProps = props

    const theme = React.useContext<BackyardTheme>(ThemeContext)

    if (validate && process.env.NODE_ENV !== 'production') {
        React.useEffect(() => {
            validateBackyardTheme(theme, name)
        }, [theme])
    }

    return theme
}

interface UseBackyardThemeProps {
    /**
     * Whether or not to validate component
     */
    validate?: boolean
    /**
     * Name of the component to validate
     */
    name?: string
}

export { useBackyardTheme, validateBackyardTheme }

export default useBackyardTheme
