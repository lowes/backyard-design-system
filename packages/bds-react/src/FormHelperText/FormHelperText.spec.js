import * as React from 'react'

import Info from '@lowes-tech/bds-icons/InfoOutlined'

import { render } from '../../test-utils'
import { ThemeProvider } from '../ThemeProvider'

import FormHelperText from './FormHelperText'

describe('FormHelperText Snapshots', () => {
    test('FormHelperText default snapshot', () => {
        const { asFragment } = render(
            <FormHelperText>Helper Text</FormHelperText>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe('themes', () => {
        const themes = ['light', 'dark']

        themes.forEach((theme) => {
            test(`FormHeading ${theme} theme snapshot`, () => {
                const { asFragment } = render(
                    <ThemeProvider theme={theme}>
                        <FormHelperText>Helper Text</FormHelperText>
                    </ThemeProvider>
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })

    describe('prop: indent', () => {
        const indents = [false, true]

        indents.forEach((indent) => {
            test(`FormHeading ${indent ? indent : 'default'} snapshot`, () => {
                const { asFragment } = render(
                    <FormHelperText indent={indent}>Helper Text</FormHelperText>
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })

    describe('prop: icon', () => {
        test(`FormHeading icon snapshot`, () => {
            const { asFragment } = render(
                <FormHelperText icon={<Info />}>Helper Text</FormHelperText>
            )

            expect(asFragment()).toMatchSnapshot()
        })
    })
})
