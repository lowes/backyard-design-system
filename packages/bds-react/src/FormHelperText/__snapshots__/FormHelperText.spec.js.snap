// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHelperText Snapshots FormHelperText default snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
  font-size: var(--bds-font-tag-helper-text-mobile-font-size);
  line-height: var(--bds-font-tag-helper-text-mobile-line-height);
}

.c0 {
  margin: 0;
  padding-top: 0.25rem;
}

.c0 .typography {
  color: hsl(0,0%,0%,.60);
}

.c0 .typography .icon path {
  fill: hsl(0,0%,0%,.60);
}

.c0 .icon {
  vertical-align: text-top;
}

.c0 .helper-text-icon {
  margin-right: 0.5rem;
}

.c0.c0.indent .typography {
  padding-left: var(--helper-text-indent);
}

.c0.c0.disabled .typography {
  color: hsl(0,0%,0%,.30);
}

.c0.c0.disabled .typography .icon path {
  fill: hsl(0,0%,0%,.30);
}

.c0.error .typography {
  color: hsl(350,73%,35%,1);
}

.c0.error .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.success .typography {
  color: hsl(142,86%,28%,1);
}

.c0.success .typography .icon path {
  fill: hsl(142,86%,28%,1);
}

.c0.warning .typography {
  color: hsl(350,73%,35%,1);
}

.c0.warning .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.info .typography {
  color: hsl(213,84%,45%,1);
}

.c0.info .typography .icon path {
  fill: hsl(213,84%,45%,1);
}

@media only screen and (min-width:66rem) {
  .c1 {
    font-size: var(--bds-font-tag-helper-text-desktop-font-size);
    line-height: var(--bds-font-tag-helper-text-desktop-line-height);
  }
}

<span
    class="c0 helper-text indent"
  >
    <span
      class="c1 typography variant--helper align--left"
      color="text_primary"
      style="padding-left: 1rem;"
    >
      Helper Text
    </span>
  </span>
</DocumentFragment>
`;

exports[`FormHelperText Snapshots prop: icon FormHeading icon snapshot 1`] = `
<DocumentFragment>
  .c2 {
  width: var(--style-icon-width);
  height: var(--style-icon-height);
}

.c2:not(.color-override) g,
.c2:not(.color-override) path {
  fill: var(--style-icon-color);
}

.c2:not(.color-override) g[stroke-width],
.c2:not(.color-override) path[stroke-width] {
  stroke: var(--style-icon-color);
  fill: none;
}

.c2:not(.color-override) g[stroke-width] *,
.c2:not(.color-override) path[stroke-width] * {
  fill: none;
}

.c2.start {
  margin-right: var(--bds-sizes-size-8);
}

.c2.end {
  margin-left: var(--bds-sizes-size-8);
}

.c1 {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
  font-size: var(--bds-font-tag-helper-text-mobile-font-size);
  line-height: var(--bds-font-tag-helper-text-mobile-line-height);
}

.c0 {
  margin: 0;
  padding-top: 0.25rem;
}

.c0 .typography {
  color: hsl(0,0%,0%,.60);
}

.c0 .typography .icon path {
  fill: hsl(0,0%,0%,.60);
}

.c0 .icon {
  vertical-align: text-top;
}

.c0 .helper-text-icon {
  margin-right: 0.5rem;
}

.c0.c0.indent .typography {
  padding-left: var(--helper-text-indent);
}

.c0.c0.disabled .typography {
  color: hsl(0,0%,0%,.30);
}

.c0.c0.disabled .typography .icon path {
  fill: hsl(0,0%,0%,.30);
}

.c0.error .typography {
  color: hsl(350,73%,35%,1);
}

.c0.error .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.success .typography {
  color: hsl(142,86%,28%,1);
}

.c0.success .typography .icon path {
  fill: hsl(142,86%,28%,1);
}

.c0.warning .typography {
  color: hsl(350,73%,35%,1);
}

.c0.warning .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.info .typography {
  color: hsl(213,84%,45%,1);
}

.c0.info .typography .icon path {
  fill: hsl(213,84%,45%,1);
}

@media only screen and (min-width:66rem) {
  .c1 {
    font-size: var(--bds-font-tag-helper-text-desktop-font-size);
    line-height: var(--bds-font-tag-helper-text-desktop-line-height);
  }
}

<span
    class="c0 helper-text indent"
  >
    <span
      class="c1 typography variant--helper align--left"
      color="text_primary"
      style="padding-left: 1rem;"
    >
      <svg
        aria-hidden="true"
        class="c2 icon icon-info-outlined helper-text-icon"
        role="presentation"
        style="--style-icon-height: 0.75rem; --style-icon-width: 0.75rem; --style-icon-color: hsl(0, 0%, 0%, 1);"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M8 14c3.2824 0 6-2.7176 6-6 0-3.2765-2.7235-6-6.0059-6C4.7176 2 2 4.7235 2 8c0 3.2824 2.7235 6 6 6Zm0-1c-2.7765 0-4.9941-2.2235-4.9941-5 0-2.7706 2.2117-5 4.9882-5s5 2.2294 5.0059 5c.0059 2.7765-2.2235 5-5 5Zm-.053-7.0941c.4295 0 .7648-.3412.7648-.7647 0-.4294-.3353-.7706-.7647-.7706-.4236 0-.7648.3412-.7648.7706 0 .4235.3412.7647.7648.7647Zm1.4177 5.3294c.2412 0 .4294-.1706.4294-.4118 0-.2294-.1882-.4117-.4294-.4117h-.7294V7.4177c0-.3177-.1588-.5295-.4588-.5295h-1.106c-.241 0-.4293.1824-.4293.4118 0 .2412.1882.4118.4294.4118H7.7v2.7h-.7294c-.2412 0-.4294.1823-.4294.4117 0 .2412.1882.4118.4294.4118h2.3941Z"
          fill-rule="evenodd"
        />
      </svg>
      Helper Text
    </span>
  </span>
</DocumentFragment>
`;

exports[`FormHelperText Snapshots prop: indent FormHeading default snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
  font-size: var(--bds-font-tag-helper-text-mobile-font-size);
  line-height: var(--bds-font-tag-helper-text-mobile-line-height);
}

.c0 {
  margin: 0;
  padding-top: 0.25rem;
}

.c0 .typography {
  color: hsl(0,0%,0%,.60);
}

.c0 .typography .icon path {
  fill: hsl(0,0%,0%,.60);
}

.c0 .icon {
  vertical-align: text-top;
}

.c0 .helper-text-icon {
  margin-right: 0.5rem;
}

.c0.c0.indent .typography {
  padding-left: var(--helper-text-indent);
}

.c0.c0.disabled .typography {
  color: hsl(0,0%,0%,.30);
}

.c0.c0.disabled .typography .icon path {
  fill: hsl(0,0%,0%,.30);
}

.c0.error .typography {
  color: hsl(350,73%,35%,1);
}

.c0.error .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.success .typography {
  color: hsl(142,86%,28%,1);
}

.c0.success .typography .icon path {
  fill: hsl(142,86%,28%,1);
}

.c0.warning .typography {
  color: hsl(350,73%,35%,1);
}

.c0.warning .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.info .typography {
  color: hsl(213,84%,45%,1);
}

.c0.info .typography .icon path {
  fill: hsl(213,84%,45%,1);
}

@media only screen and (min-width:66rem) {
  .c1 {
    font-size: var(--bds-font-tag-helper-text-desktop-font-size);
    line-height: var(--bds-font-tag-helper-text-desktop-line-height);
  }
}

<span
    class="c0 helper-text"
  >
    <span
      class="c1 typography variant--helper align--left"
      color="text_primary"
    >
      Helper Text
    </span>
  </span>
</DocumentFragment>
`;

exports[`FormHelperText Snapshots prop: indent FormHeading true snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
  font-size: var(--bds-font-tag-helper-text-mobile-font-size);
  line-height: var(--bds-font-tag-helper-text-mobile-line-height);
}

.c0 {
  margin: 0;
  padding-top: 0.25rem;
}

.c0 .typography {
  color: hsl(0,0%,0%,.60);
}

.c0 .typography .icon path {
  fill: hsl(0,0%,0%,.60);
}

.c0 .icon {
  vertical-align: text-top;
}

.c0 .helper-text-icon {
  margin-right: 0.5rem;
}

.c0.c0.indent .typography {
  padding-left: var(--helper-text-indent);
}

.c0.c0.disabled .typography {
  color: hsl(0,0%,0%,.30);
}

.c0.c0.disabled .typography .icon path {
  fill: hsl(0,0%,0%,.30);
}

.c0.error .typography {
  color: hsl(350,73%,35%,1);
}

.c0.error .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.success .typography {
  color: hsl(142,86%,28%,1);
}

.c0.success .typography .icon path {
  fill: hsl(142,86%,28%,1);
}

.c0.warning .typography {
  color: hsl(350,73%,35%,1);
}

.c0.warning .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.info .typography {
  color: hsl(213,84%,45%,1);
}

.c0.info .typography .icon path {
  fill: hsl(213,84%,45%,1);
}

@media only screen and (min-width:66rem) {
  .c1 {
    font-size: var(--bds-font-tag-helper-text-desktop-font-size);
    line-height: var(--bds-font-tag-helper-text-desktop-line-height);
  }
}

<span
    class="c0 helper-text indent"
  >
    <span
      class="c1 typography variant--helper align--left"
      color="text_primary"
      style="padding-left: 1rem;"
    >
      Helper Text
    </span>
  </span>
</DocumentFragment>
`;

exports[`FormHelperText Snapshots themes FormHeading dark theme snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,100%,.87);
  font-size: var(--bds-font-tag-helper-text-mobile-font-size);
  line-height: var(--bds-font-tag-helper-text-mobile-line-height);
}

.c0 {
  margin: 0;
  padding-top: 0.25rem;
}

.c0 .typography {
  color: hsl(0,0%,100%,.60);
}

.c0 .typography .icon path {
  fill: hsl(0,0%,100%,.60);
}

.c0 .icon {
  vertical-align: text-top;
}

.c0 .helper-text-icon {
  margin-right: 0.5rem;
}

.c0.c0.indent .typography {
  padding-left: var(--helper-text-indent);
}

.c0.c0.disabled .typography {
  color: hsl(0,0%,100%,.30);
}

.c0.c0.disabled .typography .icon path {
  fill: hsl(0,0%,100%,.30);
}

.c0.error .typography {
  color: hsl(350,40%,64%,1);
}

.c0.error .typography .icon path {
  fill: hsl(350,40%,64%,1);
}

.c0.success .typography {
  color: hsl(142,37%,48%,1);
}

.c0.success .typography .icon path {
  fill: hsl(142,37%,48%,1);
}

.c0.warning .typography {
  color: hsl(350,40%,64%,1);
}

.c0.warning .typography .icon path {
  fill: hsl(350,40%,64%,1);
}

.c0.info .typography {
  color: hsl(213,73%,68%,1);
}

.c0.info .typography .icon path {
  fill: hsl(213,73%,68%,1);
}

@media only screen and (min-width:66rem) {
  .c1 {
    font-size: var(--bds-font-tag-helper-text-desktop-font-size);
    line-height: var(--bds-font-tag-helper-text-desktop-line-height);
  }
}

<span
    class="c0 helper-text indent"
  >
    <span
      class="c1 typography variant--helper align--left"
      color="text_primary"
      style="padding-left: 1rem;"
    >
      Helper Text
    </span>
  </span>
</DocumentFragment>
`;

exports[`FormHelperText Snapshots themes FormHeading light theme snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
  font-size: var(--bds-font-tag-helper-text-mobile-font-size);
  line-height: var(--bds-font-tag-helper-text-mobile-line-height);
}

.c0 {
  margin: 0;
  padding-top: 0.25rem;
}

.c0 .typography {
  color: hsl(0,0%,0%,.60);
}

.c0 .typography .icon path {
  fill: hsl(0,0%,0%,.60);
}

.c0 .icon {
  vertical-align: text-top;
}

.c0 .helper-text-icon {
  margin-right: 0.5rem;
}

.c0.c0.indent .typography {
  padding-left: var(--helper-text-indent);
}

.c0.c0.disabled .typography {
  color: hsl(0,0%,0%,.30);
}

.c0.c0.disabled .typography .icon path {
  fill: hsl(0,0%,0%,.30);
}

.c0.error .typography {
  color: hsl(350,73%,35%,1);
}

.c0.error .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.success .typography {
  color: hsl(142,86%,28%,1);
}

.c0.success .typography .icon path {
  fill: hsl(142,86%,28%,1);
}

.c0.warning .typography {
  color: hsl(350,73%,35%,1);
}

.c0.warning .typography .icon path {
  fill: hsl(350,73%,35%,1);
}

.c0.info .typography {
  color: hsl(213,84%,45%,1);
}

.c0.info .typography .icon path {
  fill: hsl(213,84%,45%,1);
}

@media only screen and (min-width:66rem) {
  .c1 {
    font-size: var(--bds-font-tag-helper-text-desktop-font-size);
    line-height: var(--bds-font-tag-helper-text-desktop-line-height);
  }
}

<span
    class="c0 helper-text indent"
  >
    <span
      class="c1 typography variant--helper align--left"
      color="text_primary"
      style="padding-left: 1rem;"
    >
      Helper Text
    </span>
  </span>
</DocumentFragment>
`;
